vulcan/data/_message.py

```diff
@@ -10,59 +10,60 @@
 
 @immutable
 class Address(Serializable):
-    """An address - "descriptor" used in the system containing the user's Login ID, his names and a hash of the data.
+    """An address - "descriptor" used in the system containing the user's Global Key, his names and a information whether the user has read the message.
 
-    :var str ~.login_id: login id
-    :var str ~.address_name: address name
-    :var str ~.address_hash: address hash
-    :var str ~.initials: initials
+    :var str ~.global_key: Global Key
+    :var str ~.name: address name
+    :var int ~.has_read: whether the user has read the message
     """
 
-    login_id: int = IntegerField(key="LoginId")
-    address_name: str = StringField(key="Address")
-    address_hash: str = StringField(key="AddressHash")
-    initials: str = StringField(key="Initials")
+    global_key: str = StringField(key="GlobalKey")
+    name: str = StringField(key="Name")
+    has_read: int = IntegerField(key="HasRead", required=False)
```

Comment:
Convert that to bool - you can see an example how to convert data in the objects on-the-fly in `_grade.py`. Make sure it reads `None` for sender addresses, and for student's address (as the recipient). Make that clear in the docstring (the "weird" behavior of this being `None`).